cmd/whitesourceExecuteScan.go

```diff
@@ -90,89 +126,183 @@ func newWhitesourceUtils() *whitesourceUtilsBundle {
 	return &utils
 }
 
+// whitesourceScan stores information about scanned projects
+type whitesourceScan struct {
+	productToken         string
+	aggregateProjectName string
+	projectVersion       string
+	scannedProjects      map[string]ws.Project
+	scanTimes            map[string]time.Time
+}
+
+func (s *whitesourceScan) init() {
+	if s.scannedProjects == nil {
+		s.scannedProjects = make(map[string]ws.Project)
+	}
+	if s.scanTimes == nil {
+		s.scanTimes = make(map[string]time.Time)
+	}
+}
+
+// appendScannedProject checks that no whitesource.Project is already contained in the list of scanned projects,
+// and appends a new whitesource.Project with the given name.
+func (s *whitesourceScan) appendScannedProject(projectName string) error {
+	s.init()
+	projectName = projectName + " - " + s.projectVersion
```

Comment:
I am confused. I would assume the projectVersion belongs to the project name. Why is projectName a parameter and projectVersion is already set before?

Reply:
Originally I had it like you would have expected (i.e. projectName parameter must already include the version), but then I thought it is bad style to duplicate the version appending code at the callers.

Reply:
I could rename the parameter to "moduleName" to make it more clear.

Reply:
Maybe I did address your question. All projects should be scanned with the same productVersion. There is code elsewhere in this step which also relies on this in order to find all projects belonging to the same scan as a sub-set of all projects in the same product. In order to enforce this, I thought it is a good idea to append the version only in one place. What I said about renaming the parameter to "moduleName" still applies.

Reply:
Renamed the parameter to "moduleName" to improve consistency of "what is what".